question17/main.js

```diff
@@ -0,0 +1,151 @@
+  if (imgData){
+    renderImgUiElement(imgData);
+    } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+
+function renderImgUiElement(imgData) {
+  renderNextBtn(imgData);
+  renderPrevBtn(imgData);
+  renderPageNumElement(imgData);
+  imgLists.appendChild(getImgListsFragment(imgData));
+}
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;  
+}
+
+//get Img Elements 
+const getImgListsFragment = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+let imgNum = 0;
+const isShowElement = document.getElementsByClassName("is-show");
+const imgList = document.getElementsByClassName("imgList");
+
+
+const renderPrevBtn = (imgData) => {
+  const prevBtn = createElementWithClassName("button", "prev");
+  prevBtn.id = 'js-prevbtn';
+  prevBtn.textContent = '◀︎';
+  imgListsWrapper.appendChild(prevBtn);
+  prevBtn.disabled = true;
+
+  prevBtn.addEventListener("click", function() {
+    imgNum -= 1;
+    if (imgNum < 0) {
+      imgNum = 0;
+    }
+    isShowElement[0].classList.remove('is-show'); 
+    imgList[imgNum].classList.add('is-show');
+    switchDisableForBtn(imgNum,imgData); 
+    setNumberOfPage(imgNum,imgData); 
+  });
+}
+
+const renderNextBtn = (imgData) => {
+  const nxtBtn = createElementWithClassName("button", "next");
+  nxtBtn.id = 'js-nextbtn';
+  nxtBtn.textContent = '▶︎';
+  imgListsWrapper.appendChild(nxtBtn);
+
+  nxtBtn.addEventListener("click", function() {
+    imgNum += 1;
+    if (imgNum >= imgData.length) {
+      imgNum = imgData.length - 1;
+    }
+    isShowElement[0].classList.remove('is-show'); 
+    imgList[imgNum].classList.add('is-show');
+    switchDisableForBtn(imgNum,imgData);
+    setNumberOfPage(imgNum,imgData); 
+  });
+}
+
+
+const setNumberOfPage = (num, imgData) => {
+  document.getElementById("js-number").textContent = `${num + 1} / ${imgData.length}`;
+}
+
+const switchDisableForBtn = (num, imgData) => {
+  const prevBtnElement = document.getElementById("js-prevbtn");
+  const nxtBtnElement = document.getElementById("js-nextbtn");
+  const lengthImg = imgData.length;
+
+  if (num !== 0) {
+    prevBtnElement.disabled = false;
+  } else {
+    prevBtnElement.disabled = true;
+  }
+
+  if (num !== lengthImg - 1) {
+    nxtBtnElement.disabled = false;
+  } else {
+    nxtBtnElement.disabled = true;
+  }
```

Comment:
IMO
You can also write like this
```suggestion
  prevBtnElement.disabled = num === 0;
  nxtBtnElement.disabled = num === lengthImg - 1;
```

Reply:
Thank you for the review! It is very smart :clap::clap: I learned a lot!!! I fixed it so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/31/commits/6fcbf8b0dc530c820423762e15cc352ec7c9c917

Reply:
I also learned a lot.

Thank you.

Reply:
> IMO You can also write like this

I also learned a lot.
Thank you.